kv/kvserver: TestTruncateLog failed

kv/kvserver.TestTruncateLog [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9629834?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9629834?buildTab=artifacts#/) on release-22.1 @ [366c8feea58d696c2ad017839b3049d45aa0b855](https://github.com/cockroachdb/cockroach/commits/366c8feea58d696c2ad017839b3049d45aa0b855):


```
=== RUN   TestTruncateLog
    test_log_scope.go:79: test logs captured to: /artifacts/tmp/_tmp/751d67000aac5f3394c2369309253f02/logTestTruncateLog2677750998
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestTruncateLog
    raft_log_queue_test.go:805: -- test log scope end --
--- FAIL: TestTruncateLog (0.80s)
=== RUN   TestTruncateLog/loosely-coupled=false
    raft_log_queue_test.go:766: requested index is unavailable due to compaction
    --- FAIL: TestTruncateLog/loosely-coupled=false (0.36s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTruncateLog.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-27053

kv/kvserver.TestTruncateLog [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9640629?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9640629?buildTab=artifacts#/) on release-22.1 @ [366c8feea58d696c2ad017839b3049d45aa0b855](https://github.com/cockroachdb/cockroach/commits/366c8feea58d696c2ad017839b3049d45aa0b855):


```
=== RUN   TestTruncateLog
    test_log_scope.go:79: test logs captured to: /artifacts/tmp/_tmp/751d67000aac5f3394c2369309253f02/logTestTruncateLog132299199
    test_log_scope.go:80: use -show-logs to present logs inline
=== CONT  TestTruncateLog
    raft_log_queue_test.go:805: -- test log scope end --
--- FAIL: TestTruncateLog (0.87s)
=== RUN   TestTruncateLog/loosely-coupled=false
    raft_log_queue_test.go:766: requested index is unavailable due to compaction
    --- FAIL: TestTruncateLog/loosely-coupled=false (0.40s)
=== RUN   TestTruncateLog/loosely-coupled=true
    raft_log_queue_test.go:766: requested index is unavailable due to compaction
    --- FAIL: TestTruncateLog/loosely-coupled=true (0.43s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTruncateLog.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/kv/kvserver/batcheval/cmd_truncate_log_test.go
+++ b/pkg/kv/kvserver/batcheval/cmd_truncate_log_test.go
@@ -18,7 +18,6 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/roachpb"
 	"github.com/cockroachdb/cockroach/pkg/settings/cluster"
 	"github.com/cockroachdb/cockroach/pkg/storage"
-	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/util/hlc"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
@@ -56,8 +55,6 @@ func readTruncStates(
 func TestTruncateLog(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
-
-	skip.WithIssue(t, 79719)
 
 	ctx := context.Background()
 	const (

--- a/pkg/kv/kvserver/raft_log_queue_test.go
+++ b/pkg/kv/kvserver/raft_log_queue_test.go
@@ -703,6 +703,9 @@ func TestSnapshotLogTruncationConstraints(t *testing.T) {
 func TestTruncateLog(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
+
+	skip.WithIssue(t, 79719)
+
 	testutils.RunTrueAndFalse(t, "loosely-coupled", func(t *testing.T, looselyCoupled bool) {
 		tc := testContext{}
 		ctx := context.Background()

